### PR TITLE
[SPARK-39034][SQL][TESTS][DOCS] Add tests for options from `to_json` and `from_json`.

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to fill up the missing test for options from JSON read/write functions.

This PR also includes some typo fixing of docs.

### Why are the changes needed?

For better testability and correct the documents.

### Does this PR introduce _any_ user-facing change?

Some typo is fixed from documents.

Test-wise, it's dev-only.

### How was this patch tested?

Added some UTs for JSON read/write options.

### CODE REVIEW

File: docs/sql-data-sources-json.md [hunk 1]
```diff
@@ -155,7 +155,7 @@ Data source options of JSON can be set via:
     <td>read</td>
   </tr>
   <tr>
-    <td><code>allowNumericLeadingZero</code></td>
+    <td><code>allowNumericLeadingZeros</code></td>
```
Comment (itholic) [inline docs/sql-data-sources-json.md:158]: This should fix the https://spark.apache.org/docs/latest/sql-data-sources-json.html#data-source-option. FYI.

**Review by HyukjinKwon — APPROVED** [review]
LGTM if tests pass. cc @tgravescs FYI

File: sql/core/src/test/scala/org/apache/spark/sql/JsonFunctionsSuite.scala [hunk 2]
```diff
@@ -132,6 +132,86 @@ class JsonFunctionsSuite extends QueryTest with SharedSparkSession {
...
+    val df = Seq("""{"str": 'World'}""").toDS()
+    val schema = new StructType().add("str", StringType)
+    val options = Map("allowSingleQuotes" -> "true")
+
+    checkAnswer(
+      df.select(from_json($"value", schema, options)),
+      Row(Row("World")) :: Nil)
+  }
+
+  test("from_json with option (allowNumericLeadingZeros)") {
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/JsonFunctionsSuite.scala:165]: Hm, this test fails.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**